### PR TITLE
Added documentation on the remainder (Rem) operator for floating points.

### LINKED ISSUE #57738: What does the % operator on floats do?
I can't find any documentation on the behavior of the [`Rem` implementation](https://doc.rust-lang.org/std/primitive.f64.html#impl-Rem%3Cf64%3E-1) for `f64` (or `f32`). Where can I find out how it precisely works, regarding rounding modes etc? Is it the IEEE 754-2008 "remainder" operation?

### PR DESCRIPTION
**Pull request by DevQps**
# Description

As has been explained in #57738 the remainder operator on floating points is not clear.
This PR requests adds some information on how the `Rem` / remainder operator on floating points works.

Note also that this description is for both `Rem<f32> for f32` and `Rem<f64> for f64` implementations.

Ps. I wasn't really sure on how to formulate things. So please suggest changes if you have better idea's!

closes #57738

### CODE REVIEW

**Comment by cuviper** [comment]
I'm not sure this is enough to clarify the operation. [LLVM `frem`](https://llvm.org/docs/LangRef.html#frem-instruction) cites [libm `fmod`](https://en.cppreference.com/w/c/numeric/math/fmod) -- we could demonstrate that in Rust code like: `x % y == x - (x / y).trunc() * y`.

**Comment by DevQps** [comment]
@cuviper I think you're right. I will adjust it tomorrow and write a doc test as an example? Or do you think that would be overkill?

**Comment by cuviper** [comment]
As long as it's kept simple, an example sounds good to me.

**Comment by DevQps** [comment]
@cuviper What do you think about this?

File: src/libcore/ops/arith.rs [hunk 1]
```diff
@@ -537,6 +537,20 @@ rem_impl_integer! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
...
+        /// The remainder from the division of two floats.
+        ///
+        /// The remainder has the same sign as the dividend and is computed as:
+        /// `x - (x / y).trunc() * y`.
+        ///
+        /// # Examples
+        /// ```
+        /// let x: f32 = 4.0;
+        /// let y: f32 = 2.5;
+        /// let remainder = x - (x / y).trunc() * y;
```
Comment (cuviper) [inline src/libcore/ops/arith.rs]: I'd comment the expected value (1.5) for demonstration purposes. Might be nicer to choose inputs with a greater quotient too, so this doesn't look like just subtraction.

Comment (DevQps) [reply src/libcore/ops/arith.rs]: Alright! So what about 15.2 and 2.3? (Just some random numbers). Then I would also add a comment that the result is: 1.4

Comment (cuviper) [reply src/libcore/ops/arith.rs]: Random is fine, but maybe stick to base-2 numbers that float can represent exactly.

Comment (DevQps) [reply src/libcore/ops/arith.rs]: @cuviper I updated the example! What about this?

Comment (cuviper) [reply src/libcore/ops/arith.rs]: Works for me!

**Review by cuviper — APPROVED** [review]

**Pull request merged**